packages/neon_framework/packages/account_repository/test/account_repository_test.dart

```diff
@@ -0,0 +1,584 @@
+                .having((e) => e.active, 'active', equals(accountsList[1])),
+          ),
+        );
+
+        verify(() => storage.readLastAccount()).called(1);
+      });
+
+      test('ignores last active account when not in accounts', () async {
+        when(() => storage.readAccounts()).thenAnswer((_) async => credentialsList);
+        when(() => storage.readLastAccount()).thenAnswer((_) async => 'invalid');
+        await repository.loadAccounts(rememberLastUsedAccount: true);
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts, 'accounts', containsAllInOrder(accountsList))
+                .having((e) => e.active, 'active', equals(accountsList.first)),
+          ),
+        );
+
+        verify(() => storage.readLastAccount()).called(1);
+      });
+
+      test('prefers last active account over initial account', () async {
+        when(() => storage.readAccounts()).thenAnswer((_) async => credentialsList);
+        when(() => storage.readLastAccount()).thenAnswer((_) async => credentialsList[1].id);
+        await repository.loadAccounts(
+          rememberLastUsedAccount: true,
+          initialAccount: credentialsList.first.id,
+        );
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts, 'accounts', containsAllInOrder(accountsList))
+                .having((e) => e.active, 'active', equals(accountsList[1])),
+          ),
+        );
+
+        verify(() => storage.readLastAccount()).called(1);
+      });
+    });
+
+    group('getServerStatus', () {
+      test('removes active account', () async {
+        final mockResponse = _DynamiteResponseMock<core.Status, void>();
+        when(() => coreClient.getStatus()).thenAnswer((_) async => mockResponse);
+        when(() => mockResponse.body).thenReturn(_FakeStatus());
+
+        expect(
+          repository.getServerStatus(Uri()),
+          completion(isA<core.Status>()),
+        );
+
+        verify(() => coreClient.getStatus()).called(1);
+      });
+
+      test('rethrows http exceptions as `FetchStatusFailure`', () async {
+        when(() => coreClient.getStatus()).thenThrow(http.ClientException(''));
+
+        await expectLater(
+          repository.getServerStatus(Uri()),
+          throwsA(isA<FetchStatusFailure>().having((e) => e.error, 'error', isA<http.ClientException>())),
+        );
+
+        verify(() => coreClient.getStatus()).called(1);
+      });
+    });
+
+    group('loginFlowInit', () {
+      test('returns login endpoint and token', () async {
+        when(() => clientFlowLoginV2.init()).thenAnswer((_) async {
+          return DynamiteResponse(
+            200,
+            core.LoginFlowV2((b) {
+              b.login = 'https://login_url';
+              b.poll
+                ..token = 'token'
+                ..endpoint = 'token';
```

Comment:
```suggestion
                ..endpoint = 'endpoint';
```
If the implementation accidentally returned the endpoint instead of the token field this would go unnoticed.